starport/cmd/network_chain_list.go

```diff
@@ -0,0 +1,103 @@
+package starportcmd
+
+import (
+	"fmt"
+	"io"
+	"os"
+	"sync"
+
+	"github.com/olekukonko/tablewriter"
+	"github.com/spf13/cobra"
+	launchtypes "github.com/tendermint/spn/x/launch/types"
+	"github.com/tendermint/starport/starport/pkg/clispinner"
+	"github.com/tendermint/starport/starport/pkg/cosmosaccount"
+	"github.com/tendermint/starport/starport/pkg/events"
+	"github.com/tendermint/starport/starport/services/network"
+)
+
+var LaunchSummaryHeader = []string{"Launch ID", "Chain ID", "Source", "Campaign ID"}
+
+// LaunchSummary holds summarized information about a chain launch
+type LaunchSummary struct {
+	LaunchID   string
+	ChainID    string
+	Source     string
+	CampaignID string
+}
+
+// NewNetworkChainList returns a new command to list all published chains on Starport Network
+func NewNetworkChainList() *cobra.Command {
+	c := &cobra.Command{
+		Use:   "list",
+		Short: "List published chains",
+		Args:  cobra.NoArgs,
+		RunE:  networkChainListHandler,
+	}
+	c.Flags().String(flagFrom, cosmosaccount.DefaultAccount, "Account name to use for sending transactions to SPN")
+	c.Flags().AddFlagSet(flagSetKeyringBackend())
+	c.Flags().AddFlagSet(flagSetHome())
+
+	return c
+}
+
+func networkChainListHandler(cmd *cobra.Command, args []string) error {
+	// TODO: use routine from network init PR
+	s := clispinner.New()
+	defer s.Stop()
+	var (
+		wg sync.WaitGroup
+		ev = events.NewBus()
+	)
+	wg.Add(1)
+	defer wg.Wait()
+	defer ev.Shutdown()
+	go printEvents(&wg, ev, s)
+
+	nb, err := newNetwork(cmd, network.CollectEvents(ev))
+	if err != nil {
+		return err
+	}
+
+	chains, err := nb.ChainLaunches(cmd.Context())
+	if err != nil {
+		return err
+	}
+	sums := launchSummaries(chains)
+
+	s.Stop()
+	renderLaunchSummaries(sums, os.Stdout)
+
+	return nil
+}
+
+// launchSummaries returns the list of launch summaries from the list of chain launches
+func launchSummaries(chains []launchtypes.Chain) (sums []LaunchSummary) {
+	for _, chain := range chains {
+		var campaignID string
+		if chain.HasCampaign {
+			campaignID = fmt.Sprintf("%d", chain.CampaignID)
+		} else {
+			campaignID = "no campaign"
+		}
+
+		sums = append(sums, LaunchSummary{
+			LaunchID:   fmt.Sprintf("%d", chain.LaunchID),
+			ChainID:    chain.GenesisChainID,
+			Source:     chain.SourceURL,
+			CampaignID: campaignID,
+		})
+	}
+	return sums
+}
+
+// renderLaunchSummaries writes into the provided out, the list of summarized launches
+func renderLaunchSummaries(launchSummaries []LaunchSummary, out io.Writer) {
+	launchTable := tablewriter.NewWriter(out)
```

Comment:
Instead of using tablewriter pkg, can we simplify the format by using the same listing UI in `starport a list` cmd?

If that makes sense we can move the renderer we used in the `starport a list` command to a separate package under pkg/ end reuse everywhere.

I remember, we had a consensus on simplifying it this way.

Reply:
The format we have for printAccounts is here: https://github.com/tendermint/starport/blob/02a7ba83a37a59d498b631b37d8b45a9cdb2296b/starport/cmd/account.go#L42

Basically, we can generalize this UI writer to with this API:

_pkg/entrywriter_:
```go
package entrywriter

type Tab string

type Entry []interface{}

// return an error if any of the Entry's len does not match the len of tabs.
func Write(out io.Writer, tabs []Tab, entries ...Entry) error {}
```

implementation of printAccounts can be changed as:

```go
func printAccounts(cmd *cobra.Command, accounts ...cosmosaccount.Account) {
   var entries []entrywriter.Entry

   for _, acc := range accounts {
      entries = append(entries, entrywriter.Entry{
			acc.Name,
			acc.Address(getAddressPrefix(cmd)),
			acc.PubKey(),
      })
   }

   entrywriter.Write(os.Stdout, []entrywriter.Tab{"name", "address", "public key"}, entries)
}
```

and we can use entrywriter.Write here for listing chains as well.